### ISSUE TITLE
Bug: tcpdump spawned as a zombie process due to order-of-operations error

### ISSUE THREAD

**Issue by 0x27** [cuckoosandbox/cuckoo#626]
Host is Debian 8, machinery is KVM/Libvirt. All is setup according to the documentation to do parallel processing.

When I test by spinning up a VM manually and spawning tcpdump on the interface created, tcpdump works as expected. However, when cuckoo starts tcpdump, tcpdump becomes a zombie process.

With a bit of debugging I discovered the root of the problem, and would like a hand resolving it.

```
2015-08-20 12:37:37,914 [modules.auxiliary.sniffer] INFO: Started sniffer with PID 30529 (interface=vnet0, host=192.168.10.10, dump path=/home/cuckoo/cuckoo/storage/ana
lyses/29/dump.pcap)
2015-08-20 12:37:37,915 [lib.cuckoo.core.plugins] DEBUG: Started auxiliary module: Sniffer
2015-08-20 12:37:37,944 [lib.cuckoo.common.abstracts] DEBUG: Starting machine vm1-clone
2015-08-20 12:37:37,944 [lib.cuckoo.common.abstracts] DEBUG: Getting status for vm1-clone
tcpdump: vnet0: No such device exists
(SIOCGIFHWADDR: No such device)
2015-08-20 12:37:37,982 [lib.cuckoo.common.abstracts] DEBUG: Using snapshot snapshot1 for virtual machine vm1-clone
2015-08-20 12:38:24,908 [lib.cuckoo.common.abstracts] DEBUG: Getting status for vm1-clone
2015-08-20 12:38:24,927 [lib.cuckoo.core.guest] INFO: Starting analysis on guest (id=cuckoo0, ip=192.168.10.10)
2015-08-20 12:38:24,927 [lib.cuckoo.core.guest] DEBUG: cuckoo0: waiting for status 0x0001
2015-08-20 12:38:24,972 [lib.cuckoo.core.guest] DEBUG: cuckoo0: status ready
2015-08-20 12:38:25,195 [lib.cuckoo.core.guest] DEBUG: Uploading analyzer to guest (id=cuckoo0, ip=192.168.10.10)
2015-08-20 12:38:26,341 [lib.cuckoo.core.guest] DEBUG: cuckoo0: analyzer started with PID 1184
```

if the sniffer BEFORE the VM starts, which causes issues as the VM interface is not spawned by KVM until after the VM starts. This leads to tcpdump basically going "fuck all this" and bailing out.

I was wondering if anyone had hit this problem and had a fix for it?

**Comment by brad-accuvant** [comment]
Don't sniff on vnet0, sniff on virbr0 or whatever you have.

**Comment by 0x27** [comment]
Surely if I sniff on br0 (the bridge to eth2) this will mean the sniffer will pick up all of the traffic from all of the VM's running concurrently? Or does the BPF logic filter this out? (I could probably add said BPF magic if needed, actually).

Just to clarify before I go making a massive mess of things :)

**Comment by brad-accuvant** [comment]
Nah it's the other way around actually.  For each network you have set up with libvirt there will be a vnet* interface created at runtime.  Multiple VMs can share these, but each VM should have its on virbr interface, so you won't get cross-VM network traffic.

**Comment by brad-accuvant** [comment]
Hi, sorry, wanted to update you on this since I gave you some incorrect information earlier.  While sniffing on virbr0 will be fine in most cases due to the BPF applied (as you correctly noted) you'll run into problems when using Tor transparent proxying (the packets will be logged after mangling).  I've documented how to fix that case here: https://github.com/brad-accuvant/cuckoo-modified/commit/963530d57f35dc2a1b6b6b146c7cf860f321bdc9

The script I wrote up waits until the interface has been created (up to a max of 15 seconds) and then turns itself into tcpdump with all of the original arguments -- this way Cuckoo's monitoring of the sniffer didn't need to be changed at all.
